website/blog/2023-09-14-owo-1/index.md

```diff
@@ -0,0 +1,152 @@
+
+```rust
+let mut w = op.writer_with("test.txt").append(true).await?;
+w.write(content_a).await?;
+w.write(content_b).await?;
+w.close().await?;
+```
+
+This way, users can reuse the `Writer` in their own logic without handling `append` separately.
+
+#### New Lister API
+
+To improve API consistency, we've made some adjustments to our listing functions. We've added `list` and `list_with` methods that perform single operations and renamed the original `list` to `lister` and `lister_with`.
+
+```rust
+// Old API
+let lister: Lister = op.list("dir").await?;
+
+// New API
+let entries: Vec<Entry> = op.list("dir").await?;
+let lister: Vec<Entry> = op.lister("dir").await?;
+```
+
+This brings uniformity to our API offerings.
+
+#### List With Metakey
+
+To speed up list operations, OpenDAL can now fetch and store metadata during the listing process. This eliminates the need for separate metadata calls:
+
+```rust
+let entries: Vec<Entry> = op
+  .list_with("dir/")
+  .metakey(Metakey::ContentLength | Metakey::ContentType).await?;
+
+// Use the metadata directly!
+let meta: &Metadata = entries[0].metadata();
+```
+
+This makes metadata retrieval more intuitive.
+
+#### Buffered Writer
+
+We've added general buffer support to optimize writing operations.
+
+```rust
+let w = op.writer_with("path/to/file").buffer(8 * 1024 * 1024).await?
+```
+
+#### Others
+
+Other improvements in the core library can be found in our [CHANGELOG](https://github.com/apache/incubator-opendal/blob/main/CHANGELOG.md).
+
+### Bindings
+
+#### Haskell
+
+[`opendal-hs`](https://github.com/apache/incubator-opendal/tree/main/bindings/haskell) is ready for it's first release! Welcome to check it out and give us some feedback.
+
+#### Java
+
+[`opendal-java`](https://github.com/apache/incubator-opendal/tree/main/bindings/java) enabled more available services in this release, allowing user to visit services like `redis` that not enabled by default in rust core. And `opendal-java` enabled blocking layer to allow users visit services like `s3` in blocking way.
+
+Welcome to integrate `opendal-java` into your project and give us some feedback.
+
+#### New bindings!
+
+- [`opendal-cpp`](https://github.com/apache/incubator-opendal/tree/main/bindings/cpp)
+- [`opendal-dotnet`](https://github.com/apache/incubator-opendal/tree/main/bindings/dotnet)
+- [`opendal-php`](https://github.com/apache/incubator-opendal/tree/main/bindings/php)
+
+### Applications
+
+#### oay
+
+[oay](https://github.com/apache/incubator-opendal/tree/main/bin/oay) is OpenDAL Gateway that allows users to access OpenDAL services via existing protocols like `s3` and `webdav`. It works like a proxy that forwarding requests to OpenDAL services.
+
+In this release, we implement basic `webdav` support. Users can convert any storage services to a webdav server!
```

Comment:
just test over `Fs`, should we clarify here?

Reply:
I think we don't need to. We can add integration tests for oay later.